apps/angular/signal-input/src/app/user.component.ts

```diff
@@ -19,23 +20,18 @@ const ageToCategory = (age: number): Category => {
   standalone: true,
   imports: [TitleCasePipe],
   template: `
-    {{ fullName | titlecase }} plays tennis in the {{ category }} category!!
+    {{ fullName() | titlecase }} plays tennis in the {{ category() }} category!!
   `,
   host: {
     class: 'text-xl text-green-800',
   },
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserComponent implements OnChanges {
-  @Input({ required: true }) name!: string;
-  @Input() lastName?: string;
-  @Input() age?: string;
+export class UserComponent {
+  name = input.required<string>();
+  lastName = input<string>();
+  age = input(0, { transform: numberAttribute });
 
-  fullName = '';
-  category: Category = 'Junior';
-
-  ngOnChanges(): void {
-    this.fullName = `${this.name} ${this.lastName ?? ''}`;
-    this.category = ageToCategory(Number(this.age) ?? 0);
-  }
+  fullName = computed(() => `${this.name()} ${this.lastName() ?? ''}`);
```

Comment:
nit: you can give the default value inside your input. 
two benefits:
- no need to infer the type
- no need for ?? ''

Reply:
good advice, merci !

all good now?